src/replicated/helpers.ts

```diff
@@ -1,4 +1,46 @@
+import {watch} from "../index";
+import * as https from "https";
+import consts from "../consts";
+import * as path from "path";
+import * as fs from "fs";
 
 export async function allowMultiDocumentResponse(userAgent) {
   return false;
 }
+
+export function getRelease(req, sequence) {
+  return new Promise((resolve, reject) => {
+    const upstreamURL = new URL(consts.upstreamEndpoint);
+    const options = {
+      host: upstreamURL.origin,
+      path: "/market" + req.originalUrl,
+      headers: req.headers,
+    };
+    options.headers.host = upstreamURL.origin;
+
+    let yamldata;
+    https.get(options, (newRes) => {
+      newRes.on("data", (chunk) => {
+        console.log(chunk);
+        yamldata = yamldata + chunk.toString();
+      });
+
+      newRes.on("end", () => {
+        console.log("end");
+        fs.writeFileSync(path.join(consts.localPath, "releases", sequence + ".yaml"), yamldata);
+        if (!fs.existsSync(path.join(consts.localPath, "current.yaml"))) {
+          fs.copyFileSync(path.join(consts.localPath, "releases", sequence + ".yaml"), path.join(consts.localPath, "current.yaml"));
+          watch();
```

Comment:
this looks like it creates multiple watches for the same file

Reply:
It only creates a watch if current.yaml had not existed beforehand, and the watch code exits if current.yaml doesn't exist
So only one watch will be running at a time